src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManagerInitInfo.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Acuminator.Utilities.Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Acuminator.Utilities.DiagnosticSuppression
+{
+	public readonly struct SuppressionManagerInitInfo : IEquatable<SuppressionManagerInitInfo>
+	{
+		public string Path { get; }
+
+		public bool GenerateSuppressionBase { get; }
+
+		public SuppressionManagerInitInfo(string path, bool generateSuppressionBase)
+		{
+			path.ThrowOnNullOrWhiteSpace(nameof(path));
+
+			Path = path;
+			GenerateSuppressionBase = generateSuppressionBase;
+		}
+
+		public override bool Equals(object obj) => obj is SuppressionManagerInitInfo initInfo
+			? Equals(initInfo)
+			: false;
+
+		public bool Equals(SuppressionManagerInitInfo other) => 
+			GenerateSuppressionBase == other.GenerateSuppressionBase && Path == other.Path;
+
+		public override int GetHashCode()
+		{
+			int hash = 17;
+
+			unchecked
+			{
+				hash = 23 * hash + (Path?.GetHashCode() ?? 0);
```

Comment:
Path cannot be empty so there is no reason for null check here

Reply:
in general they can - you can avoid struct constructors being called for example in array of struct